hq/app/services/CacheService.scala

```diff
@@ -0,0 +1,49 @@
+package services
+
+import aws.ec2.EC2
+import com.gu.Box
+import config.Config
+import model.{AwsAccount, SGInUse, SGOpenPortsDetail}
+import play.api.inject.ApplicationLifecycle
+import play.api.{Configuration, Environment, Mode}
+import rx.lang.scala.Observable
+import utils.attempt.{FailedAttempt, Failure}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+
+class CacheService(config: Configuration, lifecycle: ApplicationLifecycle, environment: Environment)(implicit ec: ExecutionContext) {
+  private val sgsBox: Box[Map[AwsAccount, Either[FailedAttempt, List[(SGOpenPortsDetail, Set[SGInUse])]]]] = Box(Map.empty)
+  private val accounts = Config.getAwsAccounts(config)
+
+  def getAllSgs(): Map[AwsAccount, Either[FailedAttempt, List[(SGOpenPortsDetail, Set[SGInUse])]]] = sgsBox.get()
+
+  def getSgsForAccount(awsAccount: AwsAccount): Either[FailedAttempt, List[(SGOpenPortsDetail, Set[SGInUse])]] = {
+    sgsBox.get().getOrElse(
+      awsAccount,
+      Left(Failure("unable to find account data in the cache", "No security group data available", 500, Some(awsAccount.id)).attempt)
+    )
+  }
+
+  private def refreshSgsBox(): Unit = {
+    println("Started refresh of the Security Groups")
+    for {
+      allFlaggedSgs <- EC2.allFlaggedSgs(accounts)
+    } yield {
+      println("Sending the refreshed data to the Security Groups Box")
+      sgsBox.send(allFlaggedSgs.toMap)
```

Comment:
If a call fails, we'll write that failure into the cache, potentially overwriting real data. This might be what we want (always see the state of things) or it might not (we might decide serving stale is ok in case of problems). I'm not sure which is best and we shouldn't change it here, but this is something for us to consider.